tests/Functional/Driver/OCI8/ResultTest.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Functional\Driver\OCI8;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver\OCI8\Driver;
+use Doctrine\DBAL\DriverManager;
+use Doctrine\DBAL\Exception\DriverException;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Tests\TestUtil;
+
+use function ini_get;
+use function sprintf;
+
+/**
+ * @requires extension oci8
+ */
+class ResultTest extends FunctionalTestCase
+{
+    /**
+     * Database connection parameters for functional test case
+     *
+     * @var array<string,mixed>
+     */
+    private $connectionParams;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->connectionParams = TestUtil::getConnectionParams();
+
+        if ($this->connection->getDriver() instanceof Driver) {
+            return;
+        }
+
+        self::markTestSkipped('oci8 only test.');
+    }
+
+    protected function tearDown(): void
+    {
+        $this->connection->executeQuery(sprintf(
+            'DROP FUNCTION %s.test_oracle_fetch_failure',
+            $this->connectionParams['user']
+        ));
+        $this->connection->executeQuery(sprintf(
+            'DROP TYPE %s.return_numbers',
+            $this->connectionParams['user']
+        ));
+
+        parent::tearDown();
+    }
+
+    /**
+     * This test will recreate the case where a data set that is larger than the
+     * oci8 default prefetch is invalidated on the database after a fetch has begun,
+     * but before the fetch has completed.
+     *
+     * Note that this test requires 2 separate user connections so that the
+     * pipelined function can be changed mid fetch.
+     */
+    public function testTruncatedFetch(): void
+    {
+        $this->expectException(DriverException::class);
+        $this->expectErrorMessageMatches(
+            '/^An exception occurred in the driver: There was an error before all rows could be fetched. '
+            . 'Error number: 2, Error String: oci_fetch_array.*/'
+        );
+
+        // Create a pipelined funtion that returns 10 rows more than the
+        // oci8 default prefetch
+        $this->createReturnTypeNeededForPipelinedFunction();
+        $expectedTotalRowCount = (int) ini_get('oci8.default_prefetch') + 10;
+        $this->createPipelinedFunction($expectedTotalRowCount);
+
+        // Create a separate connection from that used to create/update the function
+        // This must be a different user with permissions to change the given function
+        $separateConnection = $this->createSeparateConnection([
+            'user'     => 'system',
+            'password' => 'oracle',
```

Comment:
Yeah, this test requires two separate connections so that the pipelined function can be changed mid-fetch.  The default test user is 'doctrine' for the DBAL CI tests.  This hard coded user is an existing system user on the same CI database instance.  I was trying to get 2 separate connections for the default test user "doctrine" to work but couldn't get DBAL to all those two connections to work independently.  Is there a way to create two connections using the same user credentials that can work independently with DBAL?

Otherwise, I wasn't sure how to update the documentation in the phpunit.xml.dist to indicate that just for OCI8 tests, a developer would need to add credentials for two users.

Reply:
I was able to switch to use the privileged user as the second connection